ui/signed-out/src/androidMain/kotlin/social/androiddev/signedout/signin/SignInWebView.kt

```diff
@@ -9,89 +9,97 @@
 ) {
+    val webIntent = webBrowserIntent(url, MaterialTheme.colors.primary, MaterialTheme.colors.secondary)
+    val handler = Handler(Looper.getMainLooper())
 
-    DisposableEffect(Unit) {
-        onDispose {
-            // Remove user session from WebView
-            WebStorage.getInstance().deleteAllData()
-            CookieManager.getInstance().removeAllCookies(null)
+    val launcher =
+        rememberLauncherForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode == Activity.RESULT_CANCELED) {
+                // post to a handler to wait for a redirect intent as that should supersede this
+                handler.post { onCancel() }
+            }
         }
-    }
-
-    AndroidView(
-        modifier = modifier,
-        factory = {
-            WebView(it).apply {
-                setBackgroundColor(Color.TRANSPARENT)
-
-                webViewClient = object : WebViewClient() {
 
-                    override fun onPageFinished(view: WebView?, url: String?) {
-                    }
+    OnNewIntent { intent ->
+        val redirectUrl = intent?.data?.toString()
+        if (redirectUrl != null) {
+            if (shouldCancelLoadingUrl(redirectUrl)) {
+                handler.removeCallbacksAndMessages(null)
+            } else {
+                onCancel()
+            }
+        }
+    }
 
-                    override fun onReceivedError(
-                        view: WebView,
-                        request: WebResourceRequest,
-                        error: WebResourceError
-                    ) {
-                        onWebError(error.toString())
-                    }
+    Box(Modifier.fillMaxSize()) {
+        CircularProgressIndicator(Modifier
+            .align(Alignment.Center)
+            .size(84.dp)
+        )
+    }
 
-                    override fun shouldOverrideUrlLoading(
-                        view: WebView,
-                        request: WebResourceRequest
-                    ): Boolean {
-                        return shouldOverrideUrlLoading(request.url.toString())
-                    }
+    DisposableEffect(url) {
+        launcher.launch(webIntent)
+        onDispose {
+            handler.removeCallbacksAndMessages(null)
+        }
+    }
+}
 
-                    /* overriding this deprecated method is necessary for it to work on api levels < 24 */
-                    @Suppress("OVERRIDE_DEPRECATION")
-                    override fun shouldOverrideUrlLoading(
-                        view: WebView?,
-                        urlString: String?
-                    ): Boolean {
-                        return if (urlString == null) {
-                            false
-                        } else {
-                            shouldOverrideUrlLoading(urlString)
-                        }
-                    }
+private fun webBrowserIntent(url: String, primaryColor: Color, secondaryColor: Color): Intent {
```

Comment:
This seems useful to pull out of this file and as an available API in the UI:common or more general module/package

Reply:
I'm not sure, as `CustomTabsIntent` and `Intent` is android specific

Reply:
Yes, to be clear, I meant pulled out into a common android root. But it is not req/necessary and we can look at doing this later

Reply:
I personally like to make helpers like this private to where they are used _until_ it's used in more that one place, YAGNI and all that. The downside is someone might miss it and create a duplicate, though that's not the end of the world imo.

Reply:
Yup, ,^ that's the main reason I sometimes try to put stuff like this in some place that is more discoverable. Not important right now anyway. I'll resolve thread.